test/sandbox/test/slotable_test.rb

```diff
@@ -752,4 +752,20 @@ def test_slot_names_can_start_with_call
       end
     end
   end
+
+  def test_inline_slot_html_escape_with_integer
+    render_inline InlineSlotHtmlEscapeComponent.new(heading: "lorem ipsum", counter: 12, url: "/some_url") do |component|
+      component.with_empty_state do
+        "empty state"
```

Comment:
Would it make sense for this slot to return a non-string value? Otherwise we're testing the existing behavior and not the behavior mentioned in the issue.